packages/curve-common/src/features/switch-chain/ui/ChainList.tsx

```diff
@@ -0,0 +1,75 @@
+import TextField from '@mui/material/TextField'
+import { t } from '@lingui/macro'
+import SearchIcon from '@mui/icons-material/Search'
+import MenuItem from '@mui/material/MenuItem'
+import { ChainIcon } from './ChainIcon'
+import Typography from '@mui/material/Typography'
+import { ChainOption } from './ChainSwitcher'
+import { useMemo, useState } from 'react'
+import groupBy from 'lodash/groupBy'
+import CardHeader from '@mui/material/CardHeader'
+
+enum ChainType {
+  test = 'test',
+  main = 'main',
+}
+
+export function ChainList<TChainId extends number>({
+  options,
+  onChange,
+  showTestnets,
+}: {
+  onChange: (chainId: TChainId) => void
+  options: ChainOption<TChainId>[]
+  showTestnets: boolean
+}) {
+  const [searchValue, setSearchValue] = useState('')
+  const groupedOptions = useMemo(
+    () =>
+      groupBy(
+        options.filter((o) => o.label.toLowerCase().includes(searchValue.toLowerCase())),
```

Comment:
I personally use `.toLocaleLowerCase()` everywhere I can regarding user input.

Reply:
I'm not sure what's the best here, because the `o.label` is in English :thinking: